libs/jarstore.js

```diff
@@ -0,0 +1,196 @@
+/* -*- Mode: Java; tab-width: 2; indent-tabs-mode: nil; c-basic-offset: 2 -*- */
+/* vim: set shiftwidth=2 tabstop=2 autoindent cindent expandtab: */
+
+'use strict';
+
+var JARStore = (function() {
+  var DATABASE = "JARStore";
+  var VERSION = 1;
+  var OBJECT_STORE = "files";
+
+  var database;
+  var jars = new Map();
+  var jad;
+
+  var upgrade = {
+    "0to1": function(database, transaction, next) {
+      database.createObjectStore(OBJECT_STORE);
```

Comment:
It would be better to use inline keys by specifying a *keyPath* attribute in the *optionalParameters* argument to *createObjectStore*, as fs.js and compiled-method-cache.js do. Otherwise, it's expensive to retrieve all records. I know we don't currently do that here, and perhaps we never will, but the difference is so significant, the fix so simple, and the change so painful to make after-the-fact, that it would be better to do it now.